torch/utils/cpp_extension.py

```diff
@@ -2382,18 +2621,27 @@ def _write_ninja_file(path,
                       cuda_cflags,
                       cuda_post_cflags,
                       cuda_dlink_post_cflags,
+                      sycl_cflags,
+                      sycl_post_cflags,
+                      sycl_dlink_post_cflags,
                       sources,
                       objects,
                       ldflags,
                       library_target,
-                      with_cuda) -> None:
+                      with_cuda,
+                      with_sycl) -> None:
     r"""Write a ninja file that does the desired compiling and linking.
 
     `path`: Where to write this file
     `cflags`: list of flags to pass to $cxx. Can be None.
     `post_cflags`: list of flags to append to the $cxx invocation. Can be None.
     `cuda_cflags`: list of flags to pass to $nvcc. Can be None.
-    `cuda_postflags`: list of flags to append to the $nvcc invocation. Can be None.
+    `cuda_post_cflags`: list of flags to append to the $nvcc invocation. Can be None.
+    `cuda_dlink_post_cflags`: list of flags to append to the $nvcc device code link invocation. Can be None.
+    `sycl_cflags`: list of flags to pass to SYCL compiler. Can be None.
+    `sycl_post_cflags`: list of flags to append to the SYCL compiler invocation. Can be None.
+    `sycl_dlink_post_cflags`: list of flags to append to the $nvcc device code link invocation. Can be Non
+e.
```

Comment:
```suggestion
    `sycl_dlink_post_cflags`: list of flags to append to the $nvcc device code link invocation. Can be None.
```

Reply:
thx. And need to change `$nvcc` to "SYCL compiler".

Reply:
fixed